src/main/java/seedu/taassist/model/session/Date.java

```diff
@@ -0,0 +1,55 @@
+package seedu.taassist.model.session;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a Session's date in TA-Assist.
+ * Guarantees: immutable.
+ */
+public class Date implements Comparable<Date> {
+
+    public static final String MESSAGE_CONSTRAINTS = "An invalid date is provided. "
+            + "Dates should be given in YYYY-MM-DD format and must be an actual date.";
+    public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("EEE, dd MMMM yyyy");
+
+    private final LocalDate value;
+
+    /**
+     * Constructs a {@code Date}.
+     *
+     * @param date A valid date.
```

Comment:
Might want to just say that it is a `localDate` object rather than a valid date since there are no other verifications done. If not, we can also be more specific by saying that it is the date where the session is created / the date of the session.